test/functional/command/config_install_test.py

```diff
@@ -758,3 +758,81 @@ def test_without_initial_remote(self):
         assert "Copying file global.conf" in c.out
         c.run("config show *")
         assert "user.myteam:myconf: myvalue" in c.out
+
+
+class TestConfigInstallPkgSettings:
+    conanfile = textwrap.dedent("""
+        import os
+        from conan import ConanFile
+        from conan.tools.files import copy
+        class Conf(ConanFile):
+            name = "myconf"
+            version = "0.1"
+            settings = "os"
+            package_type = "configuration"
+            def package(self):
+                f = "win" if self.settings.os == "Windows" else "nix"
+                copy(self, "*.conf", src=os.path.join(self.build_folder, f), dst=self.package_folder)
+            """)
+
+    @pytest.fixture()
+    def client(self):
+        c = TestClient(default_server_user=True)
+        c.save({"conanfile.py": self.conanfile,
+                "win/global.conf": "user.myteam:myconf=mywinvalue",
+                "nix/global.conf": "user.myteam:myconf=mynixvalue",
+                })
+        c.run("export-pkg . -s os=Windows")
+        c.run("export-pkg . -s os=Linux")
+        c.run("upload * -r=default -c")
+        c.run("remove * -c")
+        return c
+
+    def test_config_install_from_pkg(self, client):
+        # Now install it
+        c = client
+        c.run("config install-pkg myconf/[*] -s os=Windows")
+        assert "myconf/0.1: Downloaded package revision" in c.out
+        assert "Copying file global.conf" in c.out
+        c.run("config show *")
+        assert "user.myteam:myconf: mywinvalue" in c.out
+
+        c.run("config install-pkg myconf/[*] -s os=Linux")
+        assert "myconf/0.1: Downloaded package revision" in c.out
+        assert "Copying file global.conf" in c.out
+        c.run("config show *")
+        assert "user.myteam:myconf: mynixvalue" in c.out
+
+    def test_config_install_from_pkg_profile(self, client):
+        # Now install it
+        c = client
+        c.save({"win.profile": "[settings]\nos=Windows",
+                "nix.profile": "[settings]\nos=Linux"})
+        c.run("config install-pkg myconf/[*] -pr=win.profile")
+        assert "myconf/0.1: Downloaded package revision" in c.out
+        assert "Copying file global.conf" in c.out
+        c.run("config show *")
+        assert "user.myteam:myconf: mywinvalue" in c.out
+
+        c.run("config install-pkg myconf/[*] -pr=nix.profile")
+        assert "myconf/0.1: Downloaded package revision" in c.out
+        assert "Copying file global.conf" in c.out
+        c.run("config show *")
+        assert "user.myteam:myconf: mynixvalue" in c.out
+
+    def test_config_install_from_pkg_profile_default(self, client):
+        # Now install it
+        c = client
+        c.save_home({"profiles/default": "[settings]\nos=Windows"})
```

Comment:
Adding an extra case of not having defined a profile would be good to check that we don't break - but the test might need to be separated to skip it for configurations not in Linux/Windows

Reply:
Not sure what you mean, I have added a couple of new tests, please check to see if this is what you meant.